enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/PipelineBuilder.scala

```diff
@@ -88,6 +103,45 @@ class PipelineBuilder(slotConfigurations: SlotConfigurations, converters: Expres
           source = source.addOperator(preSorting)
           new MergeSortOperator(ordering, slots)
 
+        case plans.Aggregation(_, groupingExpressions, aggregationExpression) if groupingExpressions.isEmpty =>
```

Comment:
Hmmm... If I understand this correctly, what we'll do is to use the input morsel first to calculate the aggregate values, and then we write those values back into the same morsel. Could we not instead break the pipeline here? From what I can see, the result will be that the input-morsel will be wider than necessary (since the aggregation values cannot be written until after all the input has been consumed), and for the output, we only use the last ref-slots of the first row.

It should work but seems inefficient.

(Same thing for both these cases)

Reply:
@pontusmelke @systay If you two agree on this, please make a card to remember that optimization. I will merge this PR now before it gets stale.